packages/upload-client/src/store.js

```diff
@@ -0,0 +1,173 @@
+import { CAR } from '@ucanto/transport'
+import * as StoreCapabilities from '@web3-storage/access/capabilities/store'
+import retry, { AbortError } from 'p-retry'
+import { serviceDID, connection } from './service.js'
+import { findCapability } from './utils.js'
+import { REQUEST_RETRIES } from './constants.js'
+
+/**
+ * Store a DAG encoded as a CAR file. The issuer needs the `store/add`
+ * delegated capability.
+ *
+ * Required delegated capability proofs: `store/add`
+ *
+ * @param {import('./types').InvocationConfig} invocationConfig Configuration
+ * for the UCAN invocation. An object with `issuer` and `proofs`.
+ *
+ * The `issuer` is the signing authority that is issuing the UCAN
+ * invocation(s). It is typically the user _agent_.
+ *
+ * The `proofs` are a set of capability delegations that prove the issuer
+ * has the capability to perform the action.
+ *
+ * The issuer needs the `store/add` delegated capability.
+ * @param {Blob} car CAR file data.
+ * @param {import('./types').RequestOptions} [options]
+ * @returns {Promise<import('./types').CARLink>}
+ */
+export async function add({ issuer, proofs }, car, options = {}) {
+  const capability = findCapability(proofs, StoreCapabilities.add.can)
+  // TODO: validate blob contains CAR data
+  const bytes = new Uint8Array(await car.arrayBuffer())
+  const link = await CAR.codec.link(bytes)
+  /* c8 ignore next */
+  const conn = options.connection ?? connection
+  const result = await retry(
+    async () => {
+      return await StoreCapabilities.add
+        .invoke({
+          issuer,
+          audience: serviceDID,
+          // @ts-expect-error expects did:${string} but cap with is ${string}:${string}
+          with: capability.with,
+          nb: { link },
```

Comment:
size is now required per https://github.com/web3-storage/w3protocol/releases/tag/access-v5.0.0